src/utils/src/Math.c

```diff
@@ -0,0 +1,30 @@
+#include "Include_i.h"
+
+/**
+ * Calculates power using Divide and Conquer style - O(log exponent) complexity
+ *
+ * WARNING:
+ * The result could overflow if the base ^ power exceeds UINT64 limit.
+ * It is the callers responsibility to sanitize the values before calling this API
+ */
+UINT64 power(UINT32 base, UINT32 exponent) {
+    UINT64 result = 1;
+
+    if (base == 0) {
+        return 0;
+    }
+
+    // loop till `exponent becomes 0
+    while (exponent) {
```

Comment:
The convention we use is to check explicitly for 0. This is done to avoid few typo errors which are hard to spot while debugging/revieweing

Reply:
okay